### PR TITLE
Frameworks/vertx web

### PR DESCRIPTION
**Pull request by pmlopes**
Implemented Vert.x3-Web using both:
* mongoDB
* PostgreSQL

### CODE REVIEW

File: frameworks/Crystal/moonshine/server-redis.cr [hunk 1]
```diff
@@ -29,6 +29,11 @@ end
 private def randomWorld
   id = rand(1..ID_MAXIMUM)
   num = REDIS.get("world:" + id.to_s)
+
+  if num == Nil
+    num = 7777
+  end
```
Comment (msmith-techempower) [inline frameworks/Crystal/moonshine/server-redis.cr]: This is testing code that be removed, correct?

File: frameworks/Crystal/moonshine/setup-postgres.sh [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 
-fw_depends crystal
+curl http://dist.crystal-lang.org/apt/setup.sh | sudo bash
+
+sudo apt-get install crystal
```
Comment (msmith-techempower) [inline frameworks/Crystal/moonshine/setup-postgres.sh]: Use `fw_depends`; do not install things this way.

File: frameworks/Crystal/moonshine/setup-redis.sh [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 
-fw_depends crystal
+#fw_depends crystal
+
+curl http://dist.crystal-lang.org/apt/setup.sh | sudo bash
+
+sudo apt-get install crystal
```
Comment (msmith-techempower) [inline frameworks/Crystal/moonshine/setup-redis.sh]: Same as above - use `fw_depends`

**Comment by msmith-techempower** [comment]
Please add a line to `.travis.yml` to include the `vertx-web` folder for automatic testing, and try to get all the crystal stuff out of this pull request.

File: frameworks/Crystal/moonshine/server-redis.cr (hunk 1 above)
Comment (blee-techempower) [inline frameworks/Crystal/moonshine/server-redis.cr]: Actual problem is this guy started his own branch from a snapshot where I accidentally merged.

**Comment by pmlopes** [comment]
branches are synced so all non vertx-web stuff is now out and i added a line to .travis.yml

File: frameworks/Java/vertx-web/create.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+INSERT INTO Fortune (id, message) VALUES (3, 'After enough decimal places, nobody gives a damn.');
+INSERT INTO Fortune (id, message) VALUES (4, 'A bad random number generator: 1, 1, 1, 1, 1, 4.33e+67, 1, 1, 1');
+INSERT INTO Fortune (id, message) VALUES (5, 'A computer program does what you tell it to do, not what you want it to do.');
+INSERT INTO Fortune (id, message) VALUES (6, 'Emacs is a nice operating system, but I prefer UNIX. — Tom Christaensen');
+INSERT INTO Fortune (id, message) VALUES (7, 'Any program that runs right is obsolete.');
+INSERT INTO Fortune (id, message) VALUES (8, 'A list is only as strong as its weakest link. — Donald Knuth');
+INSERT INTO Fortune (id, message) VALUES (9, 'Feature: A bug with seniority.');
+INSERT INTO Fortune (id, message) VALUES (10, 'Computers make very fast, very accurate mistakes.');
+INSERT INTO Fortune (id, message) VALUES (11, '<script>alert("This should not be displayed in a browser alert box.");</script>');
+INSERT INTO Fortune (id, message) VALUES (12, 'フレームワークのベンチマーク');
```
Comment (msmith-techempower) [inline frameworks/Java/vertx-web/create.sql]: Why does this file exist? Surely this can be removed, yes?

**Comment by msmith-techempower** [comment]
Maybe I am being dull here, but how does `vertx-web` differ from `vertx`?

**Comment by msmith-techempower** [comment]
Aside from my questions above, it looks good and is ready to merge pending the answers.

**Comment by pmlopes** [comment]
`vertx-web` is a sub-project from `vertx` which is a toolkit for writing sophisticated modern web applications and HTTP microservices. `vertx` is quite low level while `vertx-web` offers a simple router API and many web related handlers such `Upload`, `Auth`, `TemplateRendering`, etc...

File: frameworks/Java/vertx-web/create.sql (hunk 4 above)
Comment (pmlopes) [inline frameworks/Java/vertx-web/create.sql]: all dev. files have been removed

**Comment by msmith-techempower** [comment]
Makes sense to me; thanks for the explanation.

All tests pass in Travis; LGTM!

**Pull request merged**